snapcraft/cli/store.py

```diff
@@ -341,19 +341,34 @@ def export_login(login_file: str, snaps: str, channels: str, acls: str,
     # This is sensitive-- it should only be accessible by the owner
     private_open = functools.partial(os.open, mode=0o600)
 
-    # mypy doesn't have the opener arg in its stub. Ignore its warning
-    with open(login_file, 'w', opener=private_open) as f:  # type: ignore
-        store.conf.save(config_fd=f)
+    # Support a login_file of '-', which indicates a desire to print to stdout
+    if login_file.strip() == '-':
+        echo.info("\nExported login starts on next line:")
+        store.conf.save(config_fd=sys.stdout, encode=True)
+        print()
+
+        login_location = 'and printed above'
```

Comment:
I'd find it clearer to echo "Login successfully exported and printed above." here than to split up the string like that. Or alternatively set `login_location` to the whole string and use it below.

Reply:
Sure, done.